src/muz/base/rule_properties.cpp

```diff
@@ -256,18 +258,62 @@ bool rule_properties::check_accessor(app* n) {
         if (!use_list.contains(e))
             return false;
         for (expr* parent : use_list[e]) {
-            if (!parent)
-                return false; // top-level expressions are not guarded
-            if (is_recognizer(parent)) 
-                continue;
-            if (m.is_ite(parent) && to_app(parent)->get_arg(1) == e && is_recognizer(to_app(parent)->get_arg(0))) 
+            if (!parent) { // top-level expression
```

Comment:
I think you want to process top-level expressions _after_ "recognizers" have been populated.
So 'e' to a top-level todo list and process after use_list is handled.

Reply:
On a second thought, recognizers _should not_ be global. Their validity is context sensitive. So it depends on each expression being processed. So the entire setup with global recognizers or ctrs has to change.